.github/workflows/integration_test_charm.yaml

```
@@ -412,21 +412,41 @@ jobs:
       - name: juju status
         timeout-minutes: 1
         if: ${{ success() || (failure() && steps.tests.outcome == 'failure') }}
-        run: juju status --color --relations | tee ~/logs/juju-status.txt
+        run: |
+          MODELS=$(juju models --format json | jq -r '.models[] | select(."short-name" != "controller") | ."short-name"' | sed ':a;N;$!ba;s/\n/ /g')
+          for MODEL in $MODELS
+          do
+            juju status --model="$MODEL" --color --relations | tee "$HOME/logs/juju-status-$MODEL.txt"
+            printf "\n"
+          done
```

Simplified a bit using bash array instead of string:

```suggestion
        run: |
          declare -a MODELS=($(juju models --format json | jq -r '.models[] | select(."short-name" != "controller") | ."short-name"'))
          for MODEL in ${MODELS[@]}; do
            juju status --model="${MODEL}" --color --relations | tee ~/logs/"juju-status-${MODEL}.txt"
            echo
          done
```

BTW, do we really want `--color` in the log file?

---

Updated on https://github.com/canonical/data-platform-workflows/pull/195/commits/4034978e849b81c3a7b5661054e0b8d3e168464d. Also, I had to change `declare` to `mapfile` on https://github.com/canonical/data-platform-workflows/pull/195/commits/0faaa5087a06105956ea75cc5c61916922257167 to address a complaint made by `shellcheck`.

---

`--color` isn't needed for the log file, but it shows the colors in the GitHbu output.